x/gov/codec.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package gov;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "github.com/iov-one/weave/coin/codec.proto";
+import "google/protobuf/timestamp.proto";
+
+// Electorate defines who may vote in an election. This same group can be used in many elections
+// and is stored for re-use
+message Electorate {
+    // Participants defines a list of all signatures that are allowed to participate in a vote
+    repeated Participant participants = 1 [(gogoproto.nullable) = false];
+}
+
+// Participant clubs together a signature with a weight. The greater the weight
+// the greater the power of a participant.
+message Participant {
+    bytes signature = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+    uint32 power = 2 [(gogoproto.casttype) = "Weight"];
+}
+
+// Election Rules defines how an election is run. A proposal must be voted upon via a pre-defined ruleset.
+message ElectionRules {
+    // Human readable title.
+    string title = 1;
+    // Reference to the electorate to define the group of possible voters.
+    bytes electorate_id = 2;
```

Comment:
Moving this reference to the proposal type would make the model much more flexible and robust as both electorate and rules can be managed separately and combined as needed. Immutable groups or rules would be possible. WDYT?

Reply:
Sounds good, one note here though - we should probably consider how much we'll need to reuse these. But yeah, it's a good idea anyway

Reply:
I can imagine there will be at least two cases.

Simple vote: electorate_id = 1, required_votes = 5001
Supermajority: electorate_id=1, required_votes = 6667

maybe there is an "urgent vote" with lower voting period hours (but maybe higher required_votes, or quora or such)